src/app/qgsattributetabledialog.cpp

```diff
@@ -825,6 +828,13 @@ void QgsAttributeTableDialog::mActionZoomMapToSelectedRows_triggered()
   QgisApp::instance()->mapCanvas()->zoomToSelected( mLayer );
 }
 
+void QgsAttributeTableDialog::zoomToFeature( QgsFeatureId fid )
+{
+  // Select the feature first, then zoom to it
+  mLayer->selectByIds( QgsFeatureIds() << fid );
+  QgisApp::instance()->mapCanvas()->zoomToSelected( mLayer );
```

Comment:
I don't think we should select features here -- we could instead just use:

```suggestion
  QgisApp::instance()->mapCanvas()->zoomToFeatureIds( mLayer, { fid } );
```